cmd/argoexec/commands/artifact/delete.go

```diff
@@ -9,40 +9,112 @@ import (
 
 	"github.com/spf13/cobra"
 
-	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+	"github.com/argoproj/argo-workflows/v3/cmd/argo/commands/client"
+	"github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+	workflow "github.com/argoproj/argo-workflows/v3/pkg/client/clientset/versioned"
+	wfv1alpha1 "github.com/argoproj/argo-workflows/v3/pkg/client/clientset/versioned/typed/workflow/v1alpha1"
 	executor "github.com/argoproj/argo-workflows/v3/workflow/artifacts"
 	"github.com/argoproj/argo-workflows/v3/workflow/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 func NewArtifactDeleteCommand() *cobra.Command {
 	return &cobra.Command{
 		Use:          "delete",
 		SilenceUsage: true,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			a := &wfv1.Artifact{}
-			if err := json.Unmarshal([]byte(os.Getenv(common.EnvVarArtifact)), a); err != nil {
-				return fmt.Errorf("failed to unmarshal artifact: %w", err)
-			}
 
-			drv, err := executor.NewDriver(cmd.Context(), a, &resources{})
-			if err != nil {
-				return fmt.Errorf("failed to create driver: %w", err)
-			}
+			namespace := client.Namespace()
+			clientConfig := client.GetConfig()
+
+			if podName, ok := os.LookupEnv(common.EnvVarArtifactPodName); ok {
+
+				config, err := clientConfig.ClientConfig()
+				workflowInterface := workflow.NewForConfigOrDie(config)
+				if err != nil {
+					return err
+				}
+
+				artifactGCTaskInterface := workflowInterface.ArgoprojV1alpha1().WorkflowArtifactGCTasks(namespace)
+				labelSelector := fmt.Sprintf("%s = %s", common.LabelKeyArtifactGCPodName, podName)
+
+				taskList, err := artifactGCTaskInterface.List(context.Background(), metav1.ListOptions{LabelSelector: labelSelector})
+				if err != nil {
+					return err
+				}
 
-			if err := drv.Delete(a); err != nil {
-				return fmt.Errorf("failed to delete artifact: %w", err)
+				err = deleteArtifacts(taskList, cmd, artifactGCTaskInterface)
+				if err != nil {
+					return err
+				}
 			}
 			return nil
 		},
 	}
 }
 
-type resources struct{}
+func deleteArtifacts(taskList *v1alpha1.WorkflowArtifactGCTaskList, cmd *cobra.Command, artifactGCTaskInterface wfv1alpha1.WorkflowArtifactGCTaskInterface) error {
```

Comment:
just a minor suggestion, so feel free to take it or leave it: since you're only using the cmd to get the Context from it, you may just want to pass in the Context as an argument. Reasons: 1. Context is often passed in as an argument; 2. it shows you've separated out the concern of what the actual cmd was from this function